app/src/main/java/com/kms/cura/utils/GPSTracker.java

```diff
@@ -0,0 +1,156 @@
+import android.location.LocationManager;
+import android.os.Bundle;
+import android.os.IBinder;
+import android.support.annotation.Nullable;
+
+/**
+ * Created by linhtnvo on 6/13/2016.
+ */
+public class GPSTracker extends Service implements LocationListener {
+    private final Context context;
+
+    private boolean isGPSEnabled = false;
+    private boolean isNetworkEnabled = false;
+    private boolean canGetLocation = false;
+    private boolean error = false;
+
+
+    private Location location;
+
+    private double latitude;
+    private double longitude;
+    private String errorMessage = null;
+
+    private static final long MIN_DISTANCE_CHANGE_FOR_UPDATES = 10;
+    private static final long MIN_TIME_BW_UPDATES = 1000 * 60 * 1;
+
+    protected LocationManager locationManager;
+
+    public GPSTracker(Context context) {
+        this.context = context;
+        getLocation();
+    }
+
+    public Location getLocation() {
+        try {
+            locationManager = (LocationManager) context.getSystemService(LOCATION_SERVICE);
+
+            isGPSEnabled = locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER);
+
+            isNetworkEnabled = locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER);
+
+            if (!isGPSEnabled && !isNetworkEnabled) {
+
+            } else {
+                this.canGetLocation = true;
+
+                if (isNetworkEnabled) {
+                    locationManager.requestLocationUpdates(
+                            LocationManager.NETWORK_PROVIDER,
+                            MIN_TIME_BW_UPDATES,
+                            MIN_DISTANCE_CHANGE_FOR_UPDATES, this);
+                    if (locationManager != null) {
+                        location = locationManager
+                                .getLastKnownLocation(LocationManager.NETWORK_PROVIDER);
+
+                        if (location != null) {
+
+                            latitude = location.getLatitude();
+                            longitude = location.getLongitude();
+                        }
+                    }
+
+                }
+
+                if (isGPSEnabled) {
+                    if (location == null) {
+                        locationManager.requestLocationUpdates(
+                                LocationManager.GPS_PROVIDER,
+                                MIN_TIME_BW_UPDATES,
+                                MIN_DISTANCE_CHANGE_FOR_UPDATES, this);
+
+                        if (locationManager != null) {
+                            location = locationManager.getLastKnownLocation(LocationManager.GPS_PROVIDER);
+
+                            if (location != null) {
+                                latitude = location.getLatitude();
+                                longitude = location.getLongitude();
+                            }
+                        }
+                    }
+                }
+            }
+
+        } catch (Exception e) {
```

Comment:
Don't catch Exception here and save it into errorMessage fields and error like this. You'll never know when errors happened. Instead let getLocation throws exception out